packages/mom5/package.py

```diff
@@ -14,24 +14,26 @@ class Mom5(MakefilePackage):
     homepage = "https://www.access-nri.org.au"
     git = "https://github.com/ACCESS-NRI/mom5.git"
 
-    maintainers = ["harshula"]
+    maintainers = ["harshula", "penguian"]
 
     version("master", branch="master")
+    version("access-esm1.5", branch="access-esm1.5")
 
-    variant("deterministic", default=False, description="Deterministic build.")
-    variant("type", default="ACCESS-OM", description="Build MOM5 to support a particular use case.", values=("ACCESS-CM", "ACCESS-ESM", "ACCESS-OM", "ACCESS-OM-BGC", "MOM_solo"), multi=False)
+    variant("deterministic", default=False, when="@master", description="Deterministic build.")
+    variant("type", default="ACCESS-OM", when="@master", description="Build MOM5 to support a particular use case.", values=("ACCESS-CM", "ACCESS-ESM", "ACCESS-OM", "ACCESS-OM-BGC", "MOM_solo"), multi=False)
```

Comment:
It would be better to make these conditional on when NOT `access-esm1.5`.

Reply:
I have looked at the Spack source code, documentation and issues, and haven't found a way to create the equivalent of
```
variant(name, blah, when="NOT @access-esm1.5", ...)
```
Specifically, there is no NOT for versions. See, for example https://github.com/spack/spack/issues/1975

Reply:
What happens when we do mom5@git.<tag>?

Reply:
What happens when we do `mom5@git.<tag>`?